### DISCUSSION TITLE
Internal token for CLI / API administration

### DISCUSSION THREAD

**Discussion by lonix1** [woodpecker-ci/woodpecker#2258] [Request a Feature]
### Component

server, cli

### Describe the bug

This is not strictly a bug, but a design flaw. I first asked on the [discord server](https://discord.com/channels/838698813463724034/838698813463724037/1125415904134574080).

Suppose I want to lock down my woodpecker server - I don't want random users from my gitea server to be able to log in. So I set `WOODPECKER_OPEN=false`. The docs state that in that case I need to create users manually, using the CLI.

The problem is the CLI needs an auth token, to be able to communicate with the server. But to generate such a token I must log in to the server, which I can't do. A chicken and egg problem! :baby_chick: :egg:

### System Info

```shell
I am using 'next'
```

### Additional context

Idea: have the CLI inside the server container, and allow it to communicate without an auth token.

Idea: proposed by @6543:
> gitea use an "internal token" for this tasks... we could do similar

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

**Comment by 6543** [comment]
tldr: add an internal token for cli/api administration tasks ...

... in this regards woodpecker does not have a own user-registry - we exclusively use Oauth2

**Comment by lonix1** [comment]
So you would set the token as a environment variable in the server container and also the cli container?

(Similar to what we have now with `WOODPECKER_AGENT_SECRET`.)

**Comment by anbraten** [comment]
In this PR someone wrote down a smart way to login to Woodpecker: https://github.com/earl-warren/woodpecker/blob/6d1dc73418c4c6054a200c73098aaec81f9c0d1f/.woodpecker/forgejo.yml#L35
